packages/api/src/models/types.ts

```diff
@@ -0,0 +1,596 @@
+/*
+ *  Copyright 2025 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+/**
+ * This file should create and hold types that are identical to the model types.
+ * Use Utility types or other means if not suiting purpose
+ *
+ * TODO: RelationMappings -- not optional properties but a discriminated type (see IrrigationTask)
+ *
+ * How to use:
+ *  - Keep identical to model
+ *  - Add optional ? if type can be null, not based on required status.
+ *  - If you find a type is wrong, please also look to see if the model is correct, run tests
+ *  - If the model is correct, use a utility type instead of adding optionals etc.
+ *
+ * Once models can be converted to TS merge with the model file.
+ *
+ */
+
```

Comment:
Half the lines of code changes are here. I would consider ignoring reviewing the types file, because if they aren't working it will not build! The system/rules I used to make the file are maybe of interest.

I typed everything I touched during this PR, I have found myself making a lot of placeholder types with just an id property and it seemed like a waste so I decided to just do it finally.